backend/capellacollab/projects/toolmodels/modelsources/git/github/handler.py

```diff
@@ -131,6 +140,25 @@ def get_artifact_from_job(
             artifact_response, trusted_path_to_artifact
         )
 
+    def get_last_commit_id_for_file_path(
+        self, project_id: str, file_path: str, revision: str | None
+    ) -> t.Any:
+        if revision:
+            attribute = f"rev={revision}"
+        else:
+            attribute = f"path={file_path}&rev={self.git_model.revision}"
+        response = requests.get(
+            f"{self.git_instance.api_url}/repos{project_id}/commits?{attribute}",
```

Comment:
```suggestion
        response = requests.get(
            f"{self.git_instance.api_url}/repos{project_id}/commits?path={file_path}&rev={revision or self.git_model.revision}",
            headers=self.__get_headers(self.git_model.password)
```

Why is the `file_path` ignored when revision is set?

Reply:
We should always only get last updated state of the referenced files. We have to ignore other file updated in the repository, e.g., changes of the README.